// @metaplex-foundation/mpl-hybrid/dist/src/generated/instructions/captureV1.js
"use strict";
/**
 * This code was AUTOGENERATED using the kinobi library.
 * Please DO NOT EDIT THIS FILE, instead use visitors
 * to add features, then rerun kinobi to update it.
 *
 * @see https://github.com/metaplex-foundation/kinobi
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.captureV1 = exports.getCaptureV1InstructionDataSerializer = void 0;
const mpl_toolbox_1 = require("@metaplex-foundation/mpl-toolbox");
const umi_1 = require("@metaplex-foundation/umi");
const serializers_1 = require("@metaplex-foundation/umi/serializers");
const shared_1 = require("../shared");
function getCaptureV1InstructionDataSerializer() {
    return (0, serializers_1.mapSerializer)((0, serializers_1.struct)([['discriminator', (0, serializers_1.array)((0, serializers_1.u8)(), { size: 8 })]], { description: 'CaptureV1InstructionData' }), (value) => ({
        ...value,
        discriminator: [22, 23, 128, 17, 40, 133, 224, 228],
    }));
}
exports.getCaptureV1InstructionDataSerializer = getCaptureV1InstructionDataSerializer;
// Instruction.
function captureV1(context, input) {
    // Program ID.
    const programId = context.programs.getPublicKey('mplHybrid', 'MPL4o4wMzndgh8T1NVDxELQCj5UQfYTYEkabX3wNKtb');
    // Accounts.
    const resolvedAccounts = {
        owner: {
            index: 0,
            isWritable: true,
            value: input.owner ?? null,
        },
        authority: {
            index: 1,
            isWritable: true,
            value: input.authority ?? null,
        },
        escrow: {
            index: 2,
            isWritable: true,
            value: input.escrow ?? null,
        },
        asset: {
            index: 3,
            isWritable: true,
            value: input.asset ?? null,
        },
        collection: {
            index: 4,
            isWritable: true,
            value: input.collection ?? null,
        },
        userTokenAccount: {
            index: 5,
            isWritable: true,
            value: input.userTokenAccount ?? null,
        },
        escrowTokenAccount: {
            index: 6,
            isWritable: true,
            value: input.escrowTokenAccount ?? null,
        },
        token: {
            index: 7,
            isWritable: false,
            value: input.token ?? null,
        },
        feeTokenAccount: {
            index: 8,
            isWritable: true,
            value: input.feeTokenAccount ?? null,
        },
        feeSolAccount: {
            index: 9,
            isWritable: true,
            value: input.feeSolAccount ?? null,
        },
        feeProjectAccount: {
            index: 10,
            isWritable: true,
            value: input.feeProjectAccount ?? null,
        },
        recentBlockhashes: {
            index: 11,
            isWritable: false,
            value: input.recentBlockhashes ?? null,
        },
        mplCore: {
            index: 12,
            isWritable: false,
            value: input.mplCore ?? null,
        },
        systemProgram: {
            index: 13,
            isWritable: false,
            value: input.systemProgram ?? null,
        },
        tokenProgram: {
            index: 14,
            isWritable: false,
            value: input.tokenProgram ?? null,
        },
        associatedTokenProgram: {
            index: 15,
            isWritable: false,
            value: input.associatedTokenProgram ?? null,
        },
    };
    // Default values.
    if (!resolvedAccounts.authority.value) {
        resolvedAccounts.authority.value = context.identity;
    }
    if (!resolvedAccounts.userTokenAccount.value) {
        resolvedAccounts.userTokenAccount.value = (0, mpl_toolbox_1.findAssociatedTokenPda)(context, {
            mint: (0, shared_1.expectPublicKey)(resolvedAccounts.token.value),
            owner: (0, shared_1.expectPublicKey)(resolvedAccounts.owner.value),
        });
    }
    if (!resolvedAccounts.escrowTokenAccount.value) {
        resolvedAccounts.escrowTokenAccount.value = (0, mpl_toolbox_1.findAssociatedTokenPda)(context, {
            mint: (0, shared_1.expectPublicKey)(resolvedAccounts.token.value),
            owner: (0, shared_1.expectPublicKey)(resolvedAccounts.escrow.value),
        });
    }
    if (!resolvedAccounts.feeTokenAccount.value) {
        resolvedAccounts.feeTokenAccount.value = (0, mpl_toolbox_1.findAssociatedTokenPda)(context, {
            mint: (0, shared_1.expectPublicKey)(resolvedAccounts.token.value),
            owner: (0, shared_1.expectPublicKey)(resolvedAccounts.feeProjectAccount.value),
        });
    }
    if (!resolvedAccounts.feeSolAccount.value) {
        resolvedAccounts.feeSolAccount.value = (0, umi_1.publicKey)('GjF4LqmEhV33riVyAwHwiEeAHx4XXFn2yMY3fmMigoP3');
    }
    if (!resolvedAccounts.recentBlockhashes.value) {
        resolvedAccounts.recentBlockhashes.value = (0, umi_1.publicKey)('SysvarS1otHashes111111111111111111111111111');
    }
    if (!resolvedAccounts.mplCore.value) {
        resolvedAccounts.mplCore.value = (0, umi_1.publicKey)('CoREENxT6tW1HoK8ypY1SxRMZTcVPm7R94rH4PZNhX7d');
    }
    if (!resolvedAccounts.systemProgram.value) {
        resolvedAccounts.systemProgram.value = context.programs.getPublicKey('splSystem', '11111111111111111111111111111111');
        resolvedAccounts.systemProgram.isWritable = false;
    }
    if (!resolvedAccounts.tokenProgram.value) {
        resolvedAccounts.tokenProgram.value = context.programs.getPublicKey('splToken', 'TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA');
        resolvedAccounts.tokenProgram.isWritable = false;
    }
    if (!resolvedAccounts.associatedTokenProgram.value) {
        resolvedAccounts.associatedTokenProgram.value = (0, umi_1.publicKey)('ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL');
    }
    // Accounts in order.
    const orderedAccounts = Object.values(resolvedAccounts).sort((a, b) => a.index - b.index);
    // Keys and Signers.
    const [keys, signers] = (0, shared_1.getAccountMetasAndSigners)(orderedAccounts, 'programId', programId);
    // Data.
    const data = getCaptureV1InstructionDataSerializer().serialize({});
    // Bytes Created On Chain.
    const bytesCreatedOnChain = 0;
    return (0, umi_1.transactionBuilder)([
        { instruction: { keys, programId, data }, signers, bytesCreatedOnChain },
    ]);
}
exports.captureV1 = captureV1;
