// @metaplex-foundation/mpl-hybrid/dist/src/generated/programs/mplHybrid.js
"use strict";
/**
 * This code was AUTOGENERATED using the kinobi library.
 * Please DO NOT EDIT THIS FILE, instead use visitors
 * to add features, then rerun kinobi to update it.
 *
 * @see https://github.com/metaplex-foundation/kinobi
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMplHybridProgramId = exports.getMplHybridProgram = exports.createMplHybridProgram = exports.MPL_HYBRID_PROGRAM_ID = void 0;
const errors_1 = require("../errors");
exports.MPL_HYBRID_PROGRAM_ID = 'MPL4o4wMzndgh8T1NVDxELQCj5UQfYTYEkabX3wNKtb';
function createMplHybridProgram() {
    return {
        name: 'mplHybrid',
        publicKey: exports.MPL_HYBRID_PROGRAM_ID,
        getErrorFromCode(code, cause) {
            return (0, errors_1.getMplHybridErrorFromCode)(code, this, cause);
        },
        getErrorFromName(name, cause) {
            return (0, errors_1.getMplHybridErrorFromName)(name, this, cause);
        },
        isOnCluster() {
            return true;
        },
    };
}
exports.createMplHybridProgram = createMplHybridProgram;
function getMplHybridProgram(context, clusterFilter) {
    return context.programs.get('mplHybrid', clusterFilter);
}
exports.getMplHybridProgram = getMplHybridProgram;
function getMplHybridProgramId(context, clusterFilter) {
    return context.programs.getPublicKey('mplHybrid', exports.MPL_HYBRID_PROGRAM_ID, clusterFilter);
}
exports.getMplHybridProgramId = getMplHybridProgramId;
